### ISSUE TITLE
MD057: Links with parenthesis in file path does not exist

### ISSUE THREAD

**Issue by simonljus** [rvben/rumdl#830]
### rumdl version

0.2.62

### Installation method

mise

### Bug description

A link where the file path contains parenthesis (balanced) is reported as ` [MD057] Relative link 'pathUpToThe(ClosingParenthesis' does not exist`

Workaround: Escaping the parenthesis: `%28`  for `(` and `%29` for `)`.

### Steps to reproduce

1. Create a file `file(inner).md` in the root directory 
2. Create a directory `folder(inner)`
3. Create a file 'file.md' in the directory `folder(inner)`
4. In the root directory,  create `test.md` with the following content:

```markdown
# Test cases

## Not found

[File with parenthesis not found](file(inner).md)

[Folder with parenthesis not found](folder(inner)/file.md)

## Found

[Escaping filename found](file%28inner%29.md)

[Escaping folder found](folder%28inner%29/file.md)


```
5. Run `rumdl check test.md`

### Expected behavior

Expecting:

```
Success: No issues found in 1 file
```
Links where the file path contains balanced parenthesis should exist.

### Actual behavior

Actual:

```
test.md:5:35: [MD057] Relative link 'file(inner' does not exist
test.md:7:37: [MD057] Relative link 'folder(inner' does not exist
```

### Configuration

```toml

```

### Operating system

macOS

**Comment by rvben** [comment]
Thanks for reporting this, and for the precise example. Fixed in **v0.2.70**, with the fix contributed by @DavidPandleton in #835. Thank you both.

Verified on the released binary:

```
$ rumdl check parens.md
Success: No issues found in 1 file

$ rumdl check missing.md
missing.md:3:1: [MD057] Relative link 'nope(x).md' does not exist
```

The second case is the control: a paren path that genuinely doesn't exist is still reported, and it now names the full `nope(x).md` rather than the truncated `nope(x`.
